The new /usr/include/ncurses/term.h does not expose the members of the
"typedef struct term TERMINAL" and there was code that was using those
members.

Fix taken from upstream latest mc development release.

--- mc-4.8.17/lib/tty/tty-ncurses.c.orig	2017-05-09 10:15:04.675349154 +0000
+++ mc-4.8.17/lib/tty/tty-ncurses.c	2017-05-09 10:16:44.003941095 +0000
@@ -179,6 +179,8 @@
 void
 tty_init (gboolean mouse_enable, gboolean is_xterm)
 {
+    struct termios mode;
+
     initscr ();
 
 #ifdef HAVE_ESCDELAY
@@ -194,11 +196,15 @@
     ESCDELAY = 200;
 #endif /* HAVE_ESCDELAY */
 
+    tcgetattr (STDIN_FILENO, &mode);
     /* use Ctrl-g to generate SIGINT */
-    cur_term->Nttyb.c_cc[VINTR] = CTRL ('g');   /* ^g */
+    mode.c_cc[VINTR] = CTRL ('g');      /* ^g */
     /* disable SIGQUIT to allow use Ctrl-\ key */
-    cur_term->Nttyb.c_cc[VQUIT] = NULL_VALUE;
-    tcsetattr (cur_term->Filedes, TCSANOW, &cur_term->Nttyb);
+    mode.c_cc[VQUIT] = NULL_VALUE;
+    tcsetattr (STDIN_FILENO, TCSANOW, &mode);
+
+    /* curses remembers the "in-program" modes after this call */
+    def_prog_mode ();
 
     tty_start_interrupt_key ();
 
